.github/workflows/dockerpublish.yml

```diff
@@ -66,3 +69,52 @@ jobs:
 
           docker tag image $IMAGE_ID:$VERSION
           docker push $IMAGE_ID:$VERSION
+
+  deploy-preprod:
+
+    needs: push
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Needed for proper rebase
+      - name: Update application manifest
+        run: |
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+
+          # Update image version in application manifest
+          sed -Ei "s,(- image: docker.pkg.github.com)(.*),\1/${{ github.repository }}/$IMAGE_NAME:$VERSION," k8s/app.yaml
+
+          # Show updated manifest file
+          cat k8s/app.yaml
+
+      - name: Deploy to preprod
+        uses: steebchen/kubectl@v1.0.0
+        with:
+          args: apply -f k8s/app.yaml
+
+      - name: Wait for deploy to complete
+        uses: steebchen/kubectl@v1.0.0
+        with:
+          args: -n chime rollout status deployment.v1.apps/chime
+
+      - name: Show completed deployment
+        uses: steebchen/kubectl@v1.0.0
+        with:
+          args: -n chime get deployment chime -o yaml
+
+      - name: Push manifest update
+        uses: github-actions-x/commit@v2.5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          push-branch: ${{ env.DEPLOY_PUSH_BRANCH }}
+          commit-message: Deploy version ${VERSION} to preprod
```

Comment:
Let's not mention preprod in the commit message, since the same commit is used to ship to all environments. Maybe just `chore: bump k8s manifest up CHIME ${VERSION}`

Reply:
Good point. Addressed this in 8a553ea